### ISSUE TITLE
nunit3-console crashes under Administrator

### ISSUE THREAD

**Issue by AMelacini** [nunit/nunit#2161]
Hi All

This is what I experience

- Windows 10 VM (created via HyperV)
- nunit3-console version 3.6.1
- I run tests from a cmd prompt: OK
- If I run the test with Administrator privileges, nunit3-console starts printing the expected information but then everything crashes, including the cmd prompt!

No problem on my laptop (non VM)

Any Idea?
  Regards, Alberto

**Comment by tom-dudley** [comment]
Could you provide the console output? If the console itself is crashing perhaps try piping the output to a file when running the command.

**Comment by AMelacini** [comment]
Both stderr and stdout are empty

Below I past the command line:

nunit3-console  C:\UIAutomation\AutomatedTestWorkArea\TestAssemblies\IAMCMZTests.dll --work=C:\UIAutomation\AutomatedTestWorkArea\Results --domain=Single --result=C:\UIAutomation\AutomatedTestWorkArea\Results\_Run_In_Progress\nunit_TestResults.xml;format=nunit2 --testlist=C:\UIAutomation\AutomatedTestWorkArea\CurrentTests.txt --output=C:\UIAutomation\AutomatedTestWorkArea\Results\stdout.txt --err=C:\UIAutomation\AutomatedTestWorkArea\Results\stderr.txt

On the terminal that actually crashes this is what I captured

NUnit Console Runner 3.6.1
Copyright (C) 2017 Charlie Poole

Runtime Environment
   OS Version: Microsoft Windows NT 10.0.10586.0
  CLR Version: 4.0.30319.42000

Test Files
    C:\UIAutomation\AutomatedTestWorkArea\TestAssemblies\IAMCMZTests.dll

Test Filters
    Test: Slb.UIAutomation.IAMCMZTests.MasterTestPlan.AMS._494445Water_Rate_LimitFeature.TestCase494445_Water_Rate_Limit

Thanx, Alberto

**Comment by tom-dudley** [comment]
If the terminal is still crashing and there's further output you might have not got in time, try adding ` > output.txt` to the end of the command and see if that text file gets populated

**Comment by AMelacini** [comment]
Hi Tom

The last part of my previous message is exactly that (the output from the cmd shell)

Cheers, Alberto

**Comment by JustinRChou** [comment]
It could be a window policy issue that is causing it to crash.

**Comment by tom-dudley** [comment]
@AMelacini Could you try running with `--trace=debug` and see if you get any useful log output?

**Comment by jnm2** [comment]
@AMelacini It will also really help if you can reproduce this on a totally different machine, and if you whittle down the tests you are running until there is as little as possible left so that we have a repro and can examine it.

**Comment by AMelacini** [comment]
Good Morning/Afternoon/Evening (whichever applies).

First of all, thank you all for your time.

I did turn on the -trace=Debug and that's what I got comparing NonAdmin (everything OK) and Admin (mysterious disastrous crash with the cmd prompt box disappearing).
The two files containing meaningful information are pretty much identical with the exception (as expected) that the Admin case is truncated.



InternalTrace.nnn.log

Admin
...
08:02:06.039 Debug [ 1] TestAgency: Waiting for agent {bcde130d-f019-4e15-8e81-bb2efdaba6a4} to register
08:02:06.642 Debug [ 1] TestAgency: Returning new agent {bcde130d-f019-4e15-8e81-bb2efdaba6a4}

NonAdmin
...
07:55:37.384 Debug [ 1] TestAgency: Waiting for agent {3c13557a-efb1-4fce-a61f-69710e319521} to register
07:55:37.987 Debug [ 1] TestAgency: Returning new agent {3c13557a-efb1-4fce-a61f-69710e319521}
07:56:54.050 Info  [ 1] ProcessRunner: Done running 
07:56:54.073 Info  [ 1] ProcessRunner: Unloading 
07:56:54.607 Debug [ 1] ProcessRunner: Stopping remote agent
07:56:54.736 Info  [ 1] ServiceManager: Stopping TestFilterService
07:56:54.736 Info  [ 1] ServiceManager: Stopping ResultService
07:56:54.737 Info  [ 1] ServiceManager: Stopping TestAgency
07:56:54.737 Info  [ 1] ServiceManager: Stopping DefaultTestRunnerFactory
07:56:54.738 Info  [ 1] ServiceManager: Stopping RuntimeFrameworkService
07:56:54.738 Info  [ 1] ServiceManager: Stopping ProjectService
07:56:54.738 Info  [ 1] ServiceManager: Stopping RecentFilesService
07:56:54.741 Info  [ 1] ServiceManager: Stopping DriverService
07:56:54.741 Info  [ 1] ServiceManager: Stopping ExtensionService
07:56:54.741 Info  [ 1] ServiceManager: Stopping DomainManager
07:56:54.741 Info  [ 1] ServiceManager: Stopping SettingsService

nunit-agent_nnn.log

Admin
...
08:02:06.704 Info  [ 5] DomainManager: Creating AppDomain domain-
08:02:06.709 Debug [ 5] ServiceManager: Request for service IDriverService satisfied by DriverService
08:02:06.727 Debug [ 5] ProvidedPathsAssemblyResolver: Added path c:\Program Files (x86)\NUnit.org\nunit-console\addins

NonAdmin
...
07:55:38.032 Info  [ 5] DomainManager: Creating AppDomain domain-
07:55:38.037 Debug [ 5] ServiceManager: Request for service IDriverService satisfied by DriverService
07:55:38.054 Debug [ 5] ProvidedPathsAssemblyResolver: Added path c:\Program Files (x86)\NUnit.org\nunit-console\addins
07:56:54.607 Info  [ 5] RemoteTestAgent: Stopping


If useful I can attach the entire files.

Best Regards,   Alberto
